File: backend/src/main/java/com/ssafy/ssabre/member/dto/MemberResponse.java
package com.ssafy.ssabre.member.dto;

import com.ssafy.ssabre.member.entity.Member;
import io.swagger.v3.oas.annotations.media.Schema;

import java.time.LocalDateTime;

public record MemberResponse(
        @Schema(description = "회원 ID") Long id,
        @Schema(description = "이메일") String email,
        @Schema(description = "이름") String name,
        @Schema(description = "학번") Integer studentNo,
        @Schema(description = "캠퍼스") String campus,
        @Schema(description = "기수") Integer generation,
        @Schema(description = "반") Integer classNo,
        @Schema(description = "Mattermost ID") String mattermostId,
        @Schema(description = "프로필 이미지 URL") String profileImageUrl,
        @Schema(description = "탈퇴일") LocalDateTime deletedAt,
        @Schema(description = "생성일") LocalDateTime createdAt,
        @Schema(description = "수정일") LocalDateTime updatedAt
) {
    public static MemberResponse from(Member member, String campus, Integer generation, Integer classNo) {
        return new MemberResponse(
                member.getId(),
                member.getEmail(),
                member.getName(),
                member.getStudentNo(),
                campus,
                generation,
                classNo,
                member.getMattermostId(),
                member.getProfileImageUrl(),
                member.getDeletedAt(),
                member.getCreatedAt(),
                member.getUpdatedAt()
        );
    }
}

File: android/ssabree/app/src/main/java/com/ssafy/ssabree/core/datasource/remote/model/MemberResponse.kt
package com.ssafy.ssabree.core.datasource.remote.model

import com.google.gson.annotations.SerializedName

data class MemberResponse(
    @SerializedName("id") val id: Long,
    @SerializedName("email") val email: String?,
    @SerializedName("name") val name: String?,
    @SerializedName("studentNo") val studentNo: Int?,
    @SerializedName("campus") val campus: String?,
    @SerializedName("generation") val generation: Int?,
    @SerializedName("classNo") val classNo: Int?,
    @SerializedName("mattermostId") val mattermostId: String?,
    @SerializedName("profileImageUrl") val profileImageUrl: String?,
    @SerializedName("deletedAt") val deletedAt: String?,
    @SerializedName("createdAt") val createdAt: String?,
    @SerializedName("updatedAt") val updatedAt: String?
)
